### DISCUSSION TITLE
Serverless Framework v3 beta: feedback thread

### DISCUSSION THREAD

**Discussion by mnapoli** [serverless/serverless#10242] [General]
Hey everyone, [we just announced Serverless Framework **v3 beta** on the Serverless blog](https://www.serverless.com/blog/serverless-framework-v3-beta).

The v3 beta can be installed with: `npm i serverless@pre-3`, and it includes a redesigned CLI output:

<img width="400" alt="image" src="https://user-images.githubusercontent.com/720328/139266855-7c0c15e1-9201-48b6-92f0-db6af2da4d11.png">

You can read more [in the blog post](https://www.serverless.com/blog/serverless-framework-v3-beta).

Let's use this discussion thread to **gather feedback about the new design.**

If you see any issue, have any question or feedback, post them here 👇️

**Comment by batovpasha** [comment]
Hi,
I faced the problem that when I try to upgrade from the current latest version "2.68.0" to v3 using the following command: 
`serverless upgrade --major`
(as mentioned here: https://www.serverless.com/framework/docs/guides/upgrading-v3#upgrading-to-v3), I receive the next result:
![image](https://user-images.githubusercontent.com/31747499/144640486-ccf0de91-9d74-4d69-85c1-f4445ea2ece3.png)
I installed serverless as a standalone binary so I used this command to upgrade.
Thanks for any help!

**Reply by mnapoli** [reply]
Ohh thank you for reporting that, that documentation page needs to be updated because the v3 beta is installed differently (while in beta at least). You can find up to date instructions here: https://www.serverless.com/blog/serverless-framework-v3-beta

I'll update the documentation next week!

**Reply by batovpasha** [reply]
Thank you for the reply!

**Comment by d-fay** [comment]
First off, when I install `serverless@pre-3` I'm still seeing there is still an unpatched vulnerability in the `ansi-regex` library being reported by `npm audit`.

```
# npm audit report

ansi-regex  >2.1.1 <5.0.1
Severity: moderate
 Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
fix available via `npm audit fix --force`
Will install serverless@1.54.0, which is a breaking change
node_modules/tabtab/node_modules/ansi-regex
node_modules/tabtab/node_modules/string-width/node_modules/ansi-regex
  strip-ansi  4.0.0 - 5.2.0
  Depends on vulnerable versions of ansi-regex
  node_modules/tabtab/node_modules/string-width/node_modules/strip-ansi
  node_modules/tabtab/node_modules/strip-ansi
    inquirer  3.2.0 - 7.0.4
    Depends on vulnerable versions of string-width
    Depends on vulnerable versions of strip-ansi
    node_modules/tabtab/node_modules/inquirer
      tabtab  >=3.0.0-beta
      Depends on vulnerable versions of inquirer
      node_modules/tabtab
        serverless  >=1.55.0
        Depends on vulnerable versions of tabtab
        node_modules/serverless
    string-width  2.1.0 - 4.1.0
    Depends on vulnerable versions of strip-ansi
    node_modules/tabtab/node_modules/string-width

6 moderate severity vulnerabilities
```

Additional discussion here: https://github.com/serverless/serverless/issues/10182 
The more recent conversation on [this](https://github.com/serverless/serverless/issues/10182) issue initially had me believing that this vulnerability set would be resolved in v3.

**Reply by mnapoli** [reply]
FYI as reported in the issue you linked, this is not yet resolved in v3: https://github.com/serverless/serverless/issues/10182#issuecomment-993261241

**Comment by d-fay** [comment]
Some weird behavior here after globally installing `serverless@pre-3` globally, then locally. 

When installed globally and outside of project directory running `sls --version` returns the expected output:
```
Framework Core: 3.0.0-pre.7ec1d95c
Plugin: 6.0.0-pre.7d6455ab
SDK: 4.3.0
```

...but after creating the package json file and installing locally in a project directory, I run `sls --version` and output is missing:
```
Running "serverless" from node_modules

```

**Reply by mnapoli** [reply]
Thank you for reporting, would you be able to create an issue for this? Because that looks like an actual bug here 🤔
